google/cloud/spanner/samples/postgresql_samples.cc

```diff
@@ -67,7 +67,7 @@ void AddColumn(google::cloud::spanner_admin::DatabaseAdminClient client,
   std::vector<std::string> statements = {
       R"""(
         ALTER TABLE Albums
-            ADD COLUMN MarketingBudget BIGINT
+            ADD COLUMN NumberOfSongs BIGINT
```

Comment:
I don't believe that addresses the issue.  And it also breaks the test.

Reply:
Isn't the column already defined before? Not sure if there's breaking changes on the latest library that expect the exception to be thrown if the columns actually is a duplicate. Is there anyway to run the test locally without using gcb?

Reply:
Not sure if this is related but at least the proto definition says that it should return [`ALREADY_EXIST`](https://github.com/googleapis/googleapis/blob/6379d5fe706781af6682447f77f20d18b4db05b2/google/spanner/admin/database/v1/spanner_database_admin.proto#L573) so I guess this is expected behaviour and the test should expect to throw an exception for the same column name

Reply:
> Isn't the column already defined before?

No.  That's the problem.  Otherwise it would happen all the time.

It's presence seems to be an artifact of the `UpdateDatabaseDdl()` being replayed behind the scenes, but we are yet to understand why/when that happens, or how you're supposed to deal with it.

> Is there anyway to run the test locally without using gcb?

Yes, you should be able to build/run the test locally, although you'll need a Spanner instance to point it at.

Reply:
> Not sure if this is related but at least the proto definition says that it should return [`ALREADY_EXIST`](https://github.com/googleapis/googleapis/blob/6379d5fe706781af6682447f77f20d18b4db05b2/google/spanner/admin/database/v1/spanner_database_admin.proto#L573) so I guess this is expected behaviour and the test should expect to throw an exception for the same column name

That is correct.  But in our case it fails (very rarely) even though the column does not exist before sending the RPC, and even though we send the RPC only once.  We suspect a bug in the back end, but do not know for sure yet (and yes, the back end team has been informed).

>  Is there anyway to run the test locally without using gcb?

https://github.com/googleapis/google-cloud-cpp/blob/main/doc/contributor/howto-guide-running-ci-builds-locally.md